libsrc/atari2600/crt0.s

```diff
@@ -0,0 +1,46 @@
+; Atari VCS 2600 startup code for cc65
+;
+; Florent Flament (contact@florentflament.com), 2017
+
+        .export         _exit
+        .export         __STARTUP__ : absolute = 1
+
+        .import         __RAM_START__, __RAM_SIZE__
+        .import         copydata
+        .import         _main
+
+        .include        "zeropage.inc"
+
+
+.segment "STARTUP"
+start:
+; Clear decimal mode
+        cld
+; Clear zeropage & registers
+; Also Initializes system stack pointer
+        ldx     #0
```

Comment:
I agree this loop is a bit tricky. Nothing obvious, but with your first approach, you are missing 1 pha (After the loop the stack pointer's value is 0 while it should be $ff), which would led to using 1 additional byte of ROM for the same result. And for the second approach, you consume more bytes and the stack pointer's value is still 0.